src/modules/private-share-folder/private-sharing.controller.ts

```diff
@@ -12,12 +19,39 @@ import { Folder } from '../folder/folder.domain';
 import { User } from '../user/user.domain';
 import { OrderBy } from 'src/common/order.type';
 import { Pagination } from 'src/lib/pagination';
+import { GrantPrivilegesDto } from './dto/grant-privilages.dto';
 
 @ApiTags('Private Sharing')
 @Controller('private-sharing')
 export class PrivateSharingController {
   constructor(private readonly privateSharingUseCase: PrivateSharingUseCase) {}
 
+  @Post('grant-privileges')
+  @ApiOperation({
+    summary: 'Grant privileges to a user on a folder',
+  })
+  @ApiOkResponse({ description: 'Grant privileges to a user on a folder' })
+  @ApiBearerAuth()
+  async grantPrivileges(
+    @UserDecorator() user: User,
+    @Body() dto: GrantPrivilegesDto,
+  ): Promise<Record<'message', string>> {
+    try {
+      await this.privateSharingUseCase.grantPrivileges(
+        user,
+        dto.userUuid,
+        dto.privateFolderId,
+        dto.roleId,
+      );
+
+      return {
+        message: 'Privileges granted',
+      };
+    } catch (error) {
+      throw new BadRequestException(error.message);
```

Comment:
Do not assume that this is always a bad exception error, also do not assume that the error message should be always exposed, by obvious security reasons